packages/cardano-services-client/src/TxSubmitProvider/txSubmitHttpProvider.ts

```diff
@@ -62,6 +63,10 @@ export const txSubmitHttpProvider = (config: CreateHttpProviderConfig<TxSubmitPr
                   : ProviderFailure.BadRequest;
               throw new ProviderError(failure, txSubmissionError);
             }
+
+            if (error.name === 'HandleOwnerChangeError') {
+              Object.setPrototypeOf(error, HandleOwnerChangeError);
```

Comment:
Please add a test for this

Reply:
```
it('wraps HandleOwnerChangeErrors', async () => {
          axiosMock.onPost().replyOnce(() => {
            throw handleProviderMocks.axiosError(
              new HandleOwnerChangeError(
                '$adaHandle',
                'addr1' as Cardano.PaymentAddress,
                'addr2' as Cardano.PaymentAddress
              )
            );
          });
          const provider = txSubmitHttpProvider(config);
          try {
            await provider.submitTx({ signedTransaction: emptyUintArrayAsHexString });
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
          } catch (error: any) {
            expect(error.innerError).toBeInstanceOf(HandleOwnerChangeError);
          }
        });
```
A test like this would still expect a `ProviderError` not the `HandleOwnerChangeError` .. so I am not sure how would you test it. Is there an example somewhere?

Reply:
Let's decouple this from the current PR